src/compiler.ts

```diff
@@ -6405,12 +6453,58 @@ export class Compiler extends DiagnosticEmitter {
       case ElementKind.LOCAL: {
         let local = <Local>target;
         signature = local.type.signatureReference;
+        if(local.type.is(TypeFlags.CLOSURE)) {
```

Comment:
When the closure is still in the local scope we'll have access to their closure class still. We use information in the closure class to copy in values from the local scope before the call